Document custom labels for Node.js (#35960)
DataDog/documentation: content/en/profiler/guide/isolate-outliers-in-monolithic-services.md

[1]: https://pkg.go.dev/runtime/pprof#Do
[2]: https://pkg.go.dev/gopkg.in/DataDog/dd-trace-go.v1/profiler#WithCustomProfilerLabelKeys
[3]: https://pkg.go.dev/github.com/DataDog/dd-trace-go/v2/profiler#WithCustomProfilerLabelKeys
{{< /programming-lang >}}
{{< programming-lang lang="nodejs">}}

The Node.js profiler supports custom labels for your business logic as of `dd-trace` version 5.97.0 on Node.js 24 or later.

To add labels, use `tracer.profiling.runWithLabels()`:

```javascript
const tracer = require('dd-trace').init()

tracer.profiling.runWithLabels({ customer_name: <value> }, () => {
  /* customer-specific logic here */
})
```

> All wall time and CPU time samples taken during the function's execution, including across async continuations, carry the custom labels. Nested `runWithLabels` calls merge labels, with inner values taking precedence for duplicate keys.

Reviewer: ```suggestion
All wall time and CPU time samples taken during the function's execution, including across async continuations, carry the custom labels. Nested `runWithLabels` calls merge labels. Inner values taking precedence for duplicate keys.
```

> To specify which label keys you want to use for filtering, call `setCustomLabelKeys()` once before calling `runWithLabels()`:

Reviewer: ```suggestion
To specify which label keys you want to use for filtering, call `setCustomLabelKeys()` one time before calling `runWithLabels()`:
```

> To specify which label keys you want to use for filtering, call `setCustomLabelKeys()` as a one-time setup before calling `runWithLabels()`:

Reviewer: ```suggestion
To specify which label keys you want to use for filtering, call `setCustomLabelKeys()` once before calling `runWithLabels()`:
```

```javascript
tracer.profiling.setCustomLabelKeys(['customer_name'])
```

> Then, open wall time profiles for your service and select the `customer_name` value you're interested in under the `CPU time by` dropdown.

Reviewer: I'm assuming both are supported since line 142 says "All wall time and CPU time samples..."

```suggestion
Then, open wall time or CPU time profiles for your service and select the `customer_name` value you're interested in under the `CPU time by` dropdown.
```
